modules/social_features/social_event/src/Entity/EventEnrollment.php

```diff
@@ -161,6 +161,17 @@ public function getAccount(): ?string {
     return $this->get('field_account')->target_id;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getAccountEntity(): ?UserInterface {
+    if ($this->get('field_account')->isEmpty()) {
+      return NULL;
+    }
+
+    return $this->get('field_account')->entity;
+  }
```

Comment:
I don't think this is inefficient loading - the field `field_account` instance [already contains](https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Entity/Plugin/DataType/EntityAdapter.php#L38) a user object as `entity` property

Reply:
I'm not sure why you point to the `EntityAdapter`? 🤔 [`field_account` is an `entity_reference`](https://github.com/goalgorilla/open_social/blob/main/modules/social_features/social_event/config/install/field.storage.event_enrollment.field_account.yml#L10). That means `$this->get('field_account')` would return [`EntityReferenceFieldItemList`](https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Field/EntityReferenceFieldItemList.php) which contains a list of [`EntityReferenceItems`s](https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Field/Plugin/Field/FieldType/EntityReferenceItem.php).

If we look at the property definition of `entity` for the `entityReferenceItem` then we see it makes a data reference definition of type `entity`. That translates to a plugin ID of `entity_reference` which is [`Drupal\Core\Entity\Plugin\DataType\EntityReference`](https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Entity/Plugin/DataType/EntityReference.php). If you inspect that method you'll see it ends up calling [`getTarget()`](https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Entity/Plugin/DataType/EntityReference.php#L69) (after some indirection of `DataReferenceBase`). This `getTarget` method will perform an entity query to load the full entity (because when the field was first loaded as part of a different entity, only the ID was loaded into the `target_id` property.

Thus the following code will perform 10 database queries:
```
$enrollments = [/** 10 enrollments */];
$accounts = [];
foreach ($enrollments as $enrollment) {
  $accounts[] = $enrollment->getAccountEntity();
}
```

Meanwhile the same can be done with a single database query:
```
$enrollments = [/** 10 enrollments */];
$account_ids = [];
foreach ($enrollments as $enrollment) {
  $account_ids[] = $this->get('field_account')->target_id; // alternatively $enrollment->getAccountEntityId();
}
$accounts = User::loadMultiple($account_ids);
```

Reply:
ok, clear 👍
But I think we need to keep the `getAccountEntity()` as it is a shortcut of `User::load($uid)`